packages/protocol/contracts/L1/libs/Lib1559.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.18;
+
+import {AddressResolver} from "../../common/AddressResolver.sol";
+import {LibMath} from "../../libs/LibMath.sol";
+import {TaikoData} from "../TaikoData.sol";
+
+library Lib1559 {
+    using LibMath for uint256;
+
+    function getGasFeeStatus(
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        uint256 gasPurchaseAmount
+    )
+        internal
+        view
+        returns (
+            uint256 basefeePerGas,
+            uint256 gasPurchaseCost,
+            uint256 maxGasPurchaseAmount
+        )
+    {
+        (basefeePerGas, gasPurchaseCost, ) = purchaseGas(
+            config,
+            state.gasExcess,
+            gasPurchaseAmount
+        );
+        maxGasPurchaseAmount = getMaxGasPurchaseAmount(state, config);
+    }
+
+    function getMaxGasPurchaseAmount(
+        TaikoData.State storage state,
+        TaikoData.Config memory config
+    ) internal view returns (uint256 amount) {
+        amount = config.blockGasThrottle;
+
+        if (state.lastProposedHeight == block.number) {
+            amount -= state.gasSoldThisBlock;
+        }
+    }
+
+    function purchaseGas(
+        TaikoData.Config memory config,
+        uint256 gasExcess,
+        uint256 gasPurchaseAmount
+    )
+        internal
+        pure
+        returns (
+            uint64 basefeePerGas,
+            uint256 gasPurchaseCost,
+            uint256 newGasExcess
+        )
+    {
+        uint256 t = config.blockGasTarget;
```

Comment:
I think in our case this needs to be `blockGasThrottle` *not the per block gas limit on L2) because that's our actual target. How many L2 blocks it takes to get there (and how much gas used by each) doesn't seem important. L2 blocks don't really have much meaning for us.